components/Chips/src/MDCChipView.h

```diff
@@ -0,0 +1,190 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/*
+ A Material chip.
+
+ @see https://material.io/guidelines/components/chips.html for full details.
+
+ Chips are compact elements that represent an attribute, text, entity, or action.
+
+ Chips contain an optional leading image, a title label and an optional trailing accessory view.
+ They can also contain a leading image that appears only when the chip is selected.
+ */
+@interface MDCChipView : UIControl
+
+/*
+ A UIImageView that leads the title label.
+
+ Only visible when the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) IBInspectable UIImageView *imageView;
+
+/*
+ A UIImageView that leads the title label. Appears above the imageView.
+
+ This image view is typically used to show some icon that denotes the chip as selected, such as a
+ tick. If imageView has no image then the chip will require resizing when selected or deselected to
+ account for the changing visibility of selectedImageView.
+
+ Only visible when the chip is selected and the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) IBInspectable UIImageView *selectedImageView;
+
+/*
+ A UIView that trails the title label.
+
+ It will be given a size based on the value returned from sizeThatFits:.
+ */
+@property(nonatomic, strong, nullable) IBInspectable UIView *accessoryView;
+
+/*
+ The title label.
+
+ @note The title color is controlled by setTitleColor:forState:.
+ */
+@property(nonatomic, readonly, nonnull) IBInspectable UILabel *titleLabel;
+
+/*
+ Padding around the image view. Only used if the image view has a non-nil image.
+
+ The chip uses this property to determine intrinsicContentSize and sizeThatFits.
+
+ Defaults to (0, 0, 0, 0).
+ */
+@property(nonatomic, assign) UIEdgeInsets imagePadding UI_APPEARANCE_SELECTOR;
+
+/*
+ Padding around the accessory view. Only used if the accessory view is non-nil.
+
+ The chip uses this property to determine intrinsicContentSize and sizeThatFits.
+
+ Defaults to (0, 0, 0, 0).
+ */
+@property(nonatomic, assign) UIEdgeInsets accessoryPadding UI_APPEARANCE_SELECTOR;
+
+/*
+ Padding around the title.
+
+ The chip uses this property to determine intrinsicContentSize and sizeThatFits.
+
+ Defaults to (8, 12, 8, 12).
+ */
+@property(nonatomic, assign) UIEdgeInsets titlePadding UI_APPEARANCE_SELECTOR;
+
+/*
+ Indicates whether the chip should automatically update its font when the device’s
+ UIContentSizeCategory is changed.
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIContentSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ If set to YES, this button will base its text font on MDCFontTextStyleButton.
+
+ Default value is NO.
+ */
+@property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
```

Comment:
Why the mdc_ prefix?

Reply:
Thats the pattern we've used for this same property across all of MDC. @ianegordon may have some input here.